docs/AQL/master03-syntax.adoc

```diff
@@ -512,42 +517,169 @@ FROM
 
 ==== COUNT
 
-The `COUNT()` function returns the number of values of given _expression_ argument. The syntax is `COUNT([DISTINCT] expression |\*)`.
+The `COUNT` function returns the number of values of given _expression_ argument. The syntax is `COUNT([DISTINCT] expression |\*)`.
 
 The `COUNT(*)` is used to calculate the number of rows of the result set, whether or not they contain `NULL`.
-The `DISTINCT` keyword can be calculate the number of only the distinct values of _expression_.
+The `DISTINCT` keyword can be used to calculate the number of only distinct values of _expression_.
 
-If there are no matching rows, `COUNT()` returns `0`.
+If there are no matching rows, then this function returns `0`.
 Input values type can be anything, while the return type is always an Integer.
 
 ==== MIN
 
-The `MIN()` function returns the minimum value of given _expression_ argument. The syntax is `MIN(expression)`.
+The `MIN` function returns the minimum value of given _expression_ argument. The syntax is `MIN(expression)`.
 
-If there are no matching rows, `MIN()` returns `NULL`.
+If there are no matching rows, then this function returns `NULL`.
 Input values type should be either String, Date, Time, Integer of Real, and it will also determine the return type.
 
 ==== MAX
 
-The `MAX()` function returns the maximum value of given _expression_ argument. The syntax is `MAX(expression)`.
+The `MAX` function returns the maximum value of given _expression_ argument. The syntax is `MAX(expression)`.
 
-If there are no matching rows, `MAX()` returns `NULL`.
+If there are no matching rows, then this function returns `NULL`.
 Input values type should be either String, Date, Time, Integer of Real, and it will also determine the return type.
 
 ==== SUM
 
-The `SUM()` function returns the sum value of given _expression_ argument. The syntax is `SUM(expression)`.
+The `SUM` function returns the sum value of given _expression_ argument. The syntax is `SUM(expression)`.
 
-If there are no matching rows, `SUM()` returns `NULL`.
+If there are no matching rows, then this function returns `NULL`.
 Input values type should be either Integer of Real, and it will also determine the return type.
 
 ==== AVG
 
-The `AVG()` function returns the average value of given _expression_ argument. The syntax is `AVG(expression)`.
+The `AVG` function returns the average value of given _expression_ argument. The syntax is `AVG(expression)`.
 
-If there are no matching rows, `AVG()` returns `NULL`.
+If there are no matching rows, then this function returns `NULL`.
 Input values type should be either Integer of Real, and it will also determine the return type.
 
+=== String functions
+
+AQL is providing single-row functions for examining and manipulating string values.
+
+The table below shows the built-in AQL string functions:
+
+[width="100%",cols="1,5",options="header",]
+|===
+|Function       |Description
+
+|LENGTH()       |Returns the length of a string
```

Comment:
Probably we should underpin these functions with additions to the String class in Foundation Types - https://specifications.openehr.org/releases/BASE/latest/foundation_types.html#_string_class

Reply:
Yes, I looked into that initially, but I did not know what would be better, as:
- is_empty() is equivalent string = ""
- for is_integer(), as_integer() might not be necessary or applicable in a SQL language, usually they do an auto-casting, but I did not figure it out if we want the same behavior in AQL
- append() is related to AQL's CONCAT(), but if we allow '+' operator will be even better mapped
- less_than() is equivalent to '<' AQL operator

Maybe you see this differently?